docs/labs/redos.html

```diff
@@ -0,0 +1,159 @@
+<!DOCTYPE html>
+<html>
+<head>
+<meta http-equiv="X-UA-Compatible" content="IE=edge">
+<meta name="viewport" content="width=device-width, initial-scale=1">
+<link rel="stylesheet" href="https://best.openssf.org/assets/css/style.css">
+<link rel="stylesheet" href="checker.css">
+<script src="js-yaml.min.js"></script>
+<script src="checker.js"></script>
+<link rel="license" href="https://creativecommons.org/licenses/by/4.0/">
+
+<!-- See create_labs.md for how to create your own lab! -->
+
+<!-- Sample expected answer -->
+<script id="expected0" type="plain/text">
+  query('id').isLength({max:50}).matches( /^[A-Z0-9]+$/ ),
+</script>
+
+<!-- Full pattern of correct answer -->
+<script id="correct0" type="plain/text">
+\s* query \( ('id'|"id"|`id`) \)
+\. isLength \( \{ max: 50 ,? \} \)
+\. matches \( \/\^
+\[A-Z0-9\]\+\$\/ \) , \s*
+</script>
+
+
+<script id="info" type="application/yaml">
+---
+hints:
+- absent: ", $"
+  text: This is a parameter, it must end with a comma.
+- absent: |
+    query \( ["'`]id["'`] \)
+  text: "Use query() with an 'id' parameter."
+- present: |
+    query \( ["'`]id["'`] \) [^. ]
+  text: 'After query("id") use a period to invoke a verification method.'
+- present: |
+    (islength|Islength|IsLength|ISLENGTH)
+  text: >
+    JavaScript is case-sensitive. Use isLength instead of the case you have.
+- absent: isLength
+  text: "Limit the maximum length of input strings using isLength()."
+- present: |
+    isLength \( m
+  text: "You need to pass isLength() an object within {...}."
+- absent: matches
```

Comment:
They might not realize they need to use "max". I suggest adding:

~~~~
- absent: max
  text: You need to pass isLength() an object with the max parameter, e.g., isLength({max: VALUE})
~~~~